backend/app/adapter/db/user_changelog.go

```diff
@@ -0,0 +1,78 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/short-d/app/fw"
+	"github.com/short-d/short/app/adapter/db/table"
+	"github.com/short-d/short/app/entity"
+	"github.com/short-d/short/app/usecase/repository"
+)
+
+var _ repository.UserChangeLog = (*UserChangeLogSQL)(nil)
+
+type UserChangeLogSQL struct {
+	db    *sql.DB
+	timer fw.Timer
```

Comment:
Repositories are purely for data access. Timer should be located inside use case.